src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Windows.cs

```diff
@@ -539,13 +539,59 @@ internal static void CreateSymbolicLink(string path, string pathToTarget, bool i
                     Debug.Assert(!PathInternal.IsPartiallyQualified(targetPath));
                     return targetPath.ToString();
                 }
+                else if (rbSymlink.ReparseTag == Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_APPEXECLINK)
+                {
+                    success = MemoryMarshal.TryRead(bufferSpan, out Interop.Kernel32.AppExecLinkReparseBuffer rbAppExecLink);
+                    Debug.Assert(success);
+
+                    // The target file is at index 2
+                    if (rbAppExecLink.StringCount >= 3)
+                    {
+                        int stringListOffset = sizeof(Interop.Kernel32.AppExecLinkReparseBuffer);
+                        Span<char> stringList = MemoryMarshal.Cast<byte, char>(bufferSpan.Slice(stringListOffset));
+
+                        return GetAppExecLinkTarget(stringList);
+                    }
+                }
 
                 return null;
             }
             finally
             {
                 ArrayPool<byte>.Shared.Return(buffer);
             }
+
+            // Given an appexeclink reparse point string list, finds the
+            // start and end (null chars) of the 3rd string in the list.
+            static string? GetAppExecLinkTarget(ReadOnlySpan<char> stringList)
+            {
+                // Find the 2nd and 3rd null char
+                int start = -1;
+                int end = -1;
+                int count = 0;
+                for (int i = 0; i < stringList.Length; i++)
+                {
+                    if (stringList[i] == '\0')
+                    {
+                        count++;
+                        if (count == 2)
+                        {
+                            start = i + 1; // exclude null char
```

Comment:
Can we use `public static unsafe int IndexOf(ref char searchSpace, char value, int length)` from internal SpanHelpers?

Reply:
> if the buffer is corrupt such that \0 is the last char

If the AppExecLink struct was malformed from the beginning, then either the `StringCount` was incorrect, and/or the `sizeof` the struct was incorrect. I could add a check to ensure the `Slice` is not going beyond the `sizeof` the struct.

Reply:
We do different things based on how confident we are in the blob I guess. Eg., for reading stat etc files, we will throw if the file is malformed. I guess the min bar is to be memory safe - and Span is safe, so I think it's fine as is unless and until we discover otherwise. Either way is fine.